spec/controllers/shoutouts_controller_spec.rb

```diff
@@ -0,0 +1,245 @@
+                                              time_period_id: time_period.id,
+                                              rich_text: '<br>',
+                                              recipients: a_collection_including(user2.id.to_s, user3.id.to_s, user4.id.to_s)
+                                            })
+    end
+  end
+  describe 'PATCH #update' do
+    let!(:user) { create :user, opt_out: false }
+    let!(:user1) { create :user, opt_out: false }
+    let!(:time_period) { create :time_period }
+    let!(:time_period1) { create :time_period }
+    let!(:shoutout) do
+      create(:shoutout, { id: 1,
+                          user_id: user.id,
+                          time_period_id: time_period.id,
+                          rich_text: '<br>',
+                          recipients: [user2.id.to_s, user3.id.to_s, user4.id.to_s],
+                          digest: 1234567890 })
+    end
+    let!(:recipients) do
+      create(:shoutout_recipient, { shoutout_id: 1, user_id: user2[:id] })
+      create(:shoutout_recipient, { shoutout_id: 1, user_id: user3[:id] })
+      create(:shoutout_recipient, { shoutout_id: 1, user_id: user4[:id] })
+    end
+    let!(:shoutout2) do
+      create(:shoutout, { id: 2,
+                          user_id: user1.id,
+                          time_period_id: time_period.id,
+                          rich_text: '@Person1 thanks',
+                          recipients: [user5.id.to_s, user3.id.to_s, user4.id.to_s],
+                          digest: digital_signature_to_prevent_duplication({ user_id: user1.id.to_s,
+                                                                             time_period_id: time_period.id.to_s,
+                                                                             rich_text: '@Person1 thanks',
+                                                                             recipients: [user5.id.to_s, user3.id.to_s, user4.id.to_s] }) })
+    end
+    let!(:recipients2) do
+      create(:shoutout_recipient, { shoutout_id: 2, user_id: user5[:id] })
+      create(:shoutout_recipient, { shoutout_id: 2, user_id: user3[:id] })
+      create(:shoutout_recipient, { shoutout_id: 2, user_id: user4[:id] })
+    end
+    context 'with valid attributes' do
+      it 'updates shoutouts' do
+        allow(controller).to receive(:current_user).and_return(user)
+        patch :update, params: {  id: 1,
+                                  shoutout: { user_id: user.id,
+                                              time_period_id: time_period.id,
+                                              rich_text: '@Person1 thanks',
+                                              recipients: [user2.id, user3.id, user5.id] } }
+        shoutout.reload
+        expect(shoutout.rich_text).to eq('@Person1 thanks')
+        expect(ShoutoutRecipient.all.length).to be(6)
+        expect(ShoutoutRecipient.find_by(user_id: user5.id)[:user_id]).to be(user5.id)
```

Comment:
what do you check in this line? you find ShoutoutRecipient with user_id equal `user5.id` and then check that this ShoutoutRecipient.user_id eq `user5.id`

Reply:
I checked if the record was updated and if user5 appeared. If user is then records was updated